### ISSUE TITLE
Position of application icon is changing based on position of hidden files

### ISSUE THREAD

**Issue by m-kovac** [LinusU/node-appdmg#225]
Our team is experiencing weird behaviour with only application icon shown in visible area. When the hidden icons are moved **to the right** out of the visible area following happens:
1. If the icons are hidden, the icon position is as defined in `appdmg.json `
2. If user has `com.apple.finder "AppleShowAllFiles"`  set to true or simply hit the `CMD+Shift+.` then the position of icon is shifted.

We thought that this is issue with the appearing scrollbar, but the shift is too big, scrollbar is only few dozens pixels wide and shift is hundred or even more(did not measure exactly). We also tried to move icons down under the visible area. The icon moves counterintuitively to the right closer to the scrollbar. Is there something we are doing wrong in `appdmg.json` file? 

### DMG when `com.apple.finder "AppleShowAllFiles"` is `true` and and hidden icons are under the visible area
<img width="512" alt="HiddenDown" src="https://user-images.githubusercontent.com/118190208/228759806-e2f9e740-5a75-4ae2-b3ab-f8628e3ccc63.png">

Here is the content of the `appdmg.json` file
```
{
  "title": "App",
  "icon": "icon.icns",
  "background": "background.png",
  "background-color": "mintcream",
  "icon-size": 96,
  "window": {
    "position": { "x": 200, "y": 200 },
    "size": { "width": 400, "height": 400 }
  },
  "contents": [
    { "x": 200, "y": 200, "type": "file", "path": "App.app" },
    { "x": 0, "y": 600, "type": "position", "path": ".background" },
    { "x": 0, "y": 600, "type": "position", "path": ".DS_Store" },
    { "x": 0, "y": 600, "type": "position", "path": ".Trashes" },
    { "x": 0, "y": 600, "type": "position", "path": ".VolumeIcon.icns" }
  ]
}

```
### DMG when `com.apple.finder "AppleShowAllFiles"` is `true` and hidden icons are right to visible area
<img width="512" alt="HiddenRight" src="https://user-images.githubusercontent.com/118190208/228760327-e26dcd7a-beb3-45d5-a524-fdb85d548e25.png">

Here is the content of the `appdmg.json` file
```
{
  "title": "App",
  "icon": "icon.icns",
  "background": "background.png",
  "background-color": "mintcream",
  "icon-size": 96,
  "window": {
    "position": { "x": 200, "y": 200 },
    "size": { "width": 400, "height": 400 }
  },
  "contents": [
    { "x": 200, "y": 200, "type": "file", "path": "App.app" },
    { "x": 600, "y": 0, "type": "position", "path": ".background" },
    { "x": 600, "y": 0, "type": "position", "path": ".DS_Store" },
    { "x": 600, "y": 0, "type": "position", "path": ".Trashes" },
    { "x": 600, "y": 0, "type": "position", "path": ".VolumeIcon.icns" }
  ]
}
```

### DMG when `com.apple.finder "AppleShowAllFiles"` is `false`
<img width="512" alt="OK" src="https://user-images.githubusercontent.com/118190208/228764218-67a10c76-6de5-4312-ad02-cee99e067807.png">

**Comment by jozefizso** [comment]
We finally understood what the issue is.

TLDR: Icon size and position must be chosen so there is no room left around the icon for any of the hidden icons.

### 1. Only specify application in the content

Define size and position of the application only in the `contents` section. Alternatively include link to the `/Applications` folder.

Remove references to any of the hidden files.

### 2. Make app icon prominent

Based on the DMG window size you must make application icon quite large.

Finder will position visible hidden files automatically on the grid around the application icon. Your goal is to make it large enough and the space between window and icon must be less than `icon-size * 2` - this will make sure visible hidden files will not be placed to any of the visible area.

**Comment by jozefizso** [comment]
### Small application icon

When the application icon is small enough, Finder will position hidden files around it and in the visible area of the window.

As there may be background graphic with instruction for installing the app those hidden files may obscure it.

<img width="592" alt="Screen Shot 2023-05-16 at 11 09 36" src="https://github.com/LinusU/node-appdmg/assets/287778/16aed8c2-48f8-40d6-b751-797939622006">
<img width="592" alt="Screen Shot 2023-05-16 at 11 12 47" src="https://github.com/LinusU/node-appdmg/assets/287778/e7e8c58c-2ef7-4bac-886e-0daa74ad91af">

### Large application icon

Making the icon large enough will move hidden icons out of visible area.

The window size is `480`, application is positioned at `240x240` to be in the center and its size is `172`.

#### `com.apple.finder "AppleShowAllFiles" = false`
<img width="592" alt="Screen Shot 2023-05-16 at 11 14 53" src="https://github.com/LinusU/node-appdmg/assets/287778/a15062be-32af-4f92-bea8-d92a849eba53">

#### `com.apple.finder "AppleShowAllFiles" = true`

Hidden files will cause the scrollbar to apear but it won't affect the design of the window.

<img width="592" alt="Screen Shot 2023-05-16 at 11 14 57" src="https://github.com/LinusU/node-appdmg/assets/287778/7dda6447-3150-4271-89b9-455a8440ffcd">


### Sample `appdmg.json` config

```json
{
  "title": "Bunny app",
  "icon-size": 172,
  "contents": [
      { "x": 240, "y": 240, "type": "file", "path": "dist/BunnyApp.app" }
  ],
  "window": {
    "size": { "width": 480, "height": 480 }
  }
}
```

**Comment by cdwmhcc** [comment]
same
